Client/Frontend/Settings/HomePageSettingsViewController.swift

```diff
@@ -98,6 +98,16 @@ class WebPageSetting: StringSetting {
         textField.autocorrectionType = .No
     }
 
+    override func prepareValidValue(userInput value: String?) -> String? {
+        guard let value = value else {
+            return nil
+        }
+        if value.contains("://") {
+            return value
+        }
+        return "http://\(value)"
```

Comment:
That would be redundant with `isURL` below, though I wonder if we need both `settingIsValid` and `prepareValidValue`. Couldn't we just always use the latter to do both the check and the transformation, returning `nil` if invalid?